### PR TITLE
Added IDL and more timing details based on discussion in issues

### PR DESCRIPTION
**Pull request by lincolnfrog**
This PR adds a discussion to the explainer about the async promise resolution semantics and establishes IDL for the hit-test API. It also contains some additional discussion of how anchors relate to hit-testing.

### CODE REVIEW

File: explainer.md [hunk 1]
```diff
@@ -74,3 +93,27 @@ Creating an API that provides information about the real world has privacy impli
...
 In many ways these privacy implications are similar to video (camera) inputs and should be considered to have similar privacy implications. However, the implications are not necessarily equivalent. For example, users may consider their image to be more private than the layout of their room while a mesh of one's face may allow easier fingerprinting than RGB image processing.
+
+# API IDL
+```
+[SecureContext, Exposed=Window] interface XRHitResult {
+  readonly attribute Float32Array poseModelMatrix;
+};
+
+partial interface XRSession {
+  Promise<FrozenArray<XRHitResult>> requestHitTest(Float32Array origin, Float32Array direction, XRFrameOfReference frameOfReference);
```
Comment (toji) [inline explainer.md]: I think this is OK for now, but we'll want to make sure the ray input format here tracks whatever decision is made in https://github.com/immersive-web/webxr/issues/339

File: explainer.md [hunk 2]
```diff
@@ -74,3 +93,27 @@ Creating an API that provides information about the real world has privacy impli
...
+}
+```
+
+## API Details
+
+`hitTest` parameters
+*   origin - the origin of the ray as [x, y, z]
+*   direction - the direction of the ray as [x, y, z], normalized
```
Comment (toji) [inline explainer.md]: Normalization is a trivial thing to do, and relying on the user to input a normalized direction is going to be error prone. Best case scenario is they'll get a bad result and not understand why, worst case we'll have to check if the direction is normalized and error if not, in which case we might as well just normalize it ourselves. So I would advocate for speccing that we will normalize any non-zero-length direction vectors that are passed to the API. Here or anywhere else in WebXR for that matter.

**Comment by toji** [comment]
Left a few comments on minor points, but overall this looks good to me. Would love to hear input from other vendors.

File: explainer.md [hunk 3]
```diff
@@ -74,3 +93,27 @@ Creating an API that provides information about the real world has privacy impli
...
+    *   Note: We'll start with an  origin-direction pair instead of a pose as a pose overspecifies a ray and has the danger of the developer creating a malformed matrix
+*   frameOfReference - the frame of reference the ray origin/direction and hit results should be relative to. The call needs to include the frame of reference so the underlying system understands how to perform the calculation.
```
[the remark this answers is not in the document]
Comment (lincolnfrog) [reply explainer.md]: Done.

File: explainer.md (hunk 2 above)
Comment (lincolnfrog) [reply explainer.md]: Done.

File: explainer.md (hunk 1 above)
Comment (lincolnfrog) [reply explainer.md]: Will do - i'll keep tabs on that issue and update as appropriate.

**Pull request merged**